hibernate-reactive-core/src/main/java/org/hibernate/reactive/event/impl/AbstractReactiveFlushingEventListener.java

```diff
@@ -129,9 +128,10 @@ private ReactiveActionQueue actionQueue(EventSource session) {
 		IdentitySet copiedAlready = new IdentitySet( 10 );
 		//safe from concurrent modification because of how concurrentEntries() is implemented on IdentityMap
 		return CompletionStages.loop(
-				Arrays.stream( persistenceContext.reentrantSafeEntityEntries() )
-						.filter( entry -> flushable( entry.getValue() ) ),
-				entry -> cascadeOnFlush( session, entry.getValue().getPersister(), entry.getKey(), copiedAlready )
+				persistenceContext.reentrantSafeEntityEntries(),
+				entry -> flushable( entry.getValue() )
+						? cascadeOnFlush( session, entry.getValue().getPersister(), entry.getKey(), copiedAlready )
+						: voidFuture()
```

Comment:
So I'm not sure but this looks like the same story: by not filtering out the entities which aren't flushable before calling `loop()`, we create a much longer chain.

Unless I'm missing something.

Reply:
Sure, I got it the first time. No need to comment in every place I've used the same pattern :-)